peachjam/admin.py

```diff
@@ -413,13 +414,30 @@ class NewForm(self.new_document_form_mixin, form):
         return super().get_form(request, obj, **kwargs)
 
     def save_model(self, request, obj, form, change):
+
         if not change:
             obj.created_by = request.user
+
         super().save_model(request, obj, form, change)
 
     def save_related(self, request, form, formsets, change):
-        super().save_related(request, form, formsets, change)
         # after saving related models, also save this model again so that it can update fields based on related changes
+        # if date, title, or content changed, re-extract citations or if new document
+        from peachjam.models import citations_processor
+
+        alternative_names_formset = [
+            formset for formset in formsets if formset.model == AlternativeName
+        ][0]
+
+        if (
+            not change
+            or ["date", "title", "citation"] in form.changed_data
+            or alternative_names_formset.has_changed()
+        ):
```

Comment:
@longhotsummer Here is what I went with for the alternative names.

Reply:
@longhotsummer  I moved the logic to save_related where we can get the formsets.